docs/source/rbac/generate-scope-table.py

```diff
@@ -0,0 +1,125 @@
+import os
+from collections import defaultdict
+from pathlib import Path
+
+from pytablewriter import MarkdownTableWriter
+from ruamel.yaml import YAML
+
+from jupyterhub.scopes import scope_definitions
+
+HERE = os.path.abspath(os.path.dirname(__file__))
+PARENT = Path(HERE).parent.parent.absolute()
+
+
+class ScopeTableGenerator:
+    def __init__(self):
+        self.scopes = scope_definitions
+
+    @classmethod
+    def create_writer(cls, table_name, headers, values):
+        writer = MarkdownTableWriter()
+        writer.table_name = table_name
+        writer.headers = headers
+        writer.value_matrix = values
+        writer.margin = 1
+        return writer
+
+    def _get_scope_relationships(self):
+        """Returns a tuple of dictionary of all scope-subscope pairs and a list of just subscopes:
+
+        ({scope: subscope}, [subscopes])
+
+        used for creating hierarchical scope table in _parse_scopes()
+        """
+        pairs = []
+        for scope in self.scopes.keys():
+            if self.scopes[scope].get('subscopes'):
+                for subscope in self.scopes[scope]['subscopes']:
+                    pairs.append((scope, subscope))
+            else:
+                pairs.append((scope, None))
```

Comment:
This might be nitpicking but wouldn't it be simpler to use something like:
```python
for scope, data in self.scopes.items():
    subscopes = data.get('subscopes')
    if subscopes is not None:
        for subscope in subscopes:
            pairs.append((scope, subscope))
    else:
        pairs.append((scopes, None))
```

This would avoid multiple accesses to the various elements of the dictionnaries.

Reply:
Yeah, that's a fine improvement. Want to make a PR?

Reply:
Sure thing

Reply:
PR merged. Thanks!